### DISCUSSION TITLE
Problem with compile code for STM32F401

### DISCUSSION THREAD

**Discussion by brightproject** [stm32duino/Arduino_Core_STM32#2032] [Q&A]
I have problem with compile code for STM32F401 on MCUFRIEND_kbv library then use [Arduino_Core_STM32](https://github.com/stm32duino/Arduino_Core_STM32).
With core from rojercrark no problem.
What could be the problem?

**Comment by fpistm** [comment]
Hi @brightproject,
I've tried to compile for blackpill F4 and met issue with `WR_PIN `not defined.
Looking for that in the library, it seems you have to add STM32F4 support  need to be added to the library.
Mainly in the mcu_special*.h as it is done for other STM32.
Unfortunately, I will not be able to help on that, you should open an new issue to the library to get help.
